src/Microsoft.Azure.WebJobs.Extensions.Redis/StreamTrigger/RedisStreamTriggerScaleMonitor.cs

```diff
@@ -11,21 +12,91 @@ internal class RedisStreamTriggerScaleMonitor : RedisPollingTriggerBaseScaleMoni
+        public override async Task<RedisPollingTriggerBaseMetrics> GetMetricsAsync()
         {
-            var metrics = new RedisPollingTriggerBaseMetrics
+            long streamLength = await multiplexer.GetDatabase().StreamLengthAsync(key);
+            if (streamLength == 0)
             {
-                Remaining = multiplexer.GetDatabase().StreamLength(key),
+                return new RedisPollingTriggerBaseMetrics
+                {
+                    Remaining = 0,
+                    Timestamp = DateTime.UtcNow,
+                };
+            }
+
+            StreamGroupInfo[] groups = multiplexer.GetDatabase().StreamGroupInfo(key);
+            if (!groups.Any(g => g.Name == name))
+            {
+                // group doesn't exist, assume all entries in stream have not been processed
+                return new RedisPollingTriggerBaseMetrics
+                {
+                    Remaining = streamLength,
+                    Timestamp = DateTime.UtcNow,
+                };
+            }
+
+            StreamGroupInfo group = groups.First(g => g.Name == name);
+            if (multiplexer.GetServers()[0].Version >= RedisUtilities.Version70 && group.Lag.HasValue)
+            {
+                // Redis 7: Scaler gets number of remaining entries for the consumer group from XINFO GROUPS.
+                return new RedisPollingTriggerBaseMetrics
+                {
+                    Remaining = group.Lag.Value,
+                    Timestamp = DateTime.UtcNow,
+                };
+            }
+
+            StreamInfo info = await multiplexer.GetDatabase().StreamInfoAsync(key);
+            string firstId = (string)info.FirstEntry.Id;
+            string lastId = (string)info.LastEntry.Id;
+            string lastDeliveredId = group.LastDeliveredId ?? firstId;
+            if (lastId == lastDeliveredId)
+            {
+                // processed all entries in the stream
+                return new RedisPollingTriggerBaseMetrics
+                {
+                    Remaining = 0,
+                    Timestamp = DateTime.UtcNow,
+                };
+            }
+
+            // Redis 6/6.2 does not have an internal counter for remaining entries for the consumer group
+            // Estimate remaining entries using the time field from the entry ID (assuming ID="time-counter")
+            string[] firstIdSplit = firstId.Split('-');
+            string[] lastIdSplit = lastId.Split('-');
+            string[] lastDeliveredIdSplit = lastDeliveredId.Split('-');
+            long firstTimestamp = long.Parse(firstIdSplit[0]);
+            long lastTimestamp = long.Parse(lastIdSplit[0]);
+            long lastDeliveredTimestamp = long.Parse(lastDeliveredIdSplit[0]);
+
+            if (lastTimestamp == lastDeliveredTimestamp)
+            {
+                // If timestamp is the same, return counter difference
+                long lastCounter = long.Parse(lastIdSplit[1]);
```

Comment:
We shall use ulong https://github.com/redis/redis/blob/068051e378ceaea85a7fb011f1271587a17bd0e7/src/stream.h#L11-L14

```c
typedef struct streamID {
    uint64_t ms;        /* Unix time in milliseconds. */
    uint64_t seq;       /* Sequence number. */
} streamID;
```

Reply:
Updated to ulong